### ISSUE TITLE
Problem in adain implementation?

### ISSUE THREAD

**Issue by ericlearning** [christopher-beckham/hologan-pytorch#1]
Shouldn't the affine parameter in nn.InstanceNorm2D and nn.InstanceNorm3D be both False in adain as it gets affine transformed later?

**Comment by christopher-beckham** [comment]
I don't see an issue with having biases for them, though at the same time the architecture I have is based on my experiments and what I understood from the paper. Since the official code is now out (but in TensorFlow) I could certainly port that over to PyTorch.

**Comment by ericlearning** [comment]
The affine parameter doesn't have anything to do with biases, it has to do with the learnable parameters that are multiplied after the normalization. Since Adain replaces these learnable parameters into an output from the network, that's why affine = False should be used to disable the default learnable parameter.

**Comment by christopher-beckham** [comment]
Sorry, I should have been more clear with my reply. What I really meant to say was the following:

While it would be clearer for `affine=False`, I'm not sure if it makes a (significant) difference if it is set to true. If `affine=True`, we would get `norm(x)*a + b`, where `a` and `b` are the learnable scales and shift, respectively. But that entire result subsequently gets scaled and shifted _again_ by the network outputs `f_a` and `f_b`, resulting in `(norm(x)*a + b)*f_a + f_b`. So it's really a difference between that or the `affine=False` version which is `norm(x)*f_a + f_b`. Does that make sense, or have I messed up somewhere?

Thanks.
